### PR TITLE
Override sampling

### PR DESCRIPTION
**Pull request by anton-ryzhov**
Override per-application sampling with a global configuration file

### CODE REVIEW

**Comment by anton-ryzhov** [comment]
:+1:

File: kube_log_watcher/agents/scalyr.py [hunk 1]
```diff
@@ -155,6 +174,27 @@ def name(self):
...
+                continue
+
+            if (
+                ('component' in scalyr_sampling_rule)
+                and (scalyr_sampling_rule['component'] != container_data['component'])
+            ):
+                continue
+
+            if 'probability' in scalyr_sampling_rule:
+                container_id = int(container_data['container_id'], 16)
```
[the remark this answers is not in the document]
Comment (anton-ryzhov) [reply kube_log_watcher/agents/scalyr.py]: How could this be changed?

[the remark this answers is not in the document]
Comment (astarnell) [reply kube_log_watcher/agents/scalyr.py]: maybe
```
container_id = hash(container_data['container_id'])
```

[the remark this answers is not in the document]
Comment (anton-ryzhov) [reply kube_log_watcher/agents/scalyr.py]: No, `hash` is randomized across `cpython` runs. We want to have stable hash between restarts.
I updated to `crc32`

**Comment by anton-ryzhov** [comment]
:+1:

**Comment by astarnell** [comment]
:+1:

File: kube_log_watcher/agents/scalyr.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aermakov-zalando) [reply kube_log_watcher/agents/scalyr.py]: `crc32` is a very bad hash, I'd suggest using sha.

[the remark this answers is not in the document]
Comment (anton-ryzhov) [reply kube_log_watcher/agents/scalyr.py]: Why?

[the remark this answers is not in the document]
Comment (anton-ryzhov) [reply kube_log_watcher/agents/scalyr.py]: This is not a cryptographic encryption or signing. In fact there is no reason to use hashes longer than 7bit or even smaller.

**Pull request merged**